app/models/sushi.rb

```diff
@@ -187,7 +196,13 @@ def coerce_metric_types(params = {})
   module QueryParameterValidation
     extend ActiveSupport::Concern
     included do
-      attr_reader :platform
+      attr_reader :item_id, :platform
+    end
+
+    def validate_item_id(params)
+      raise Sushi::InvalidParameterValue.invalid_item_id(params[:item_id]) unless Hyrax::CounterMetric.exists?(['work_id LIKE ?', params[:item_id]])
```

Comment:
In SQL `LIKE` is less performant than `=`.

Consider `exists?(work_id: params[:item_id])`

Reply:
This should raise `Sushi::NotFoundError`